src/nvim/search.c

```diff
@@ -2411,6 +2412,18 @@ static int cls(void)
 {
   int c;
 
+  // Treat concealed characters like whitespace.
+  if (curwin->w_p_cole > 0 && curwin->w_cursor.col > 0
+      && curbuf->b_p_smc > 0 && curwin->w_cursor.col < curbuf->b_p_smc
+      && conceal_cursor_line(curwin)) {
+    syn_get_id(curwin, curwin->w_cursor.lnum, curwin->w_cursor.col,
+               false, NULL, false);
```

Comment:
TBH, I was just going by what I saw in `f_synconcealed()` in `eval.c`.  I tried my best to wrap my head around how the concealed character information was retrieved, but that seemed like the easiest way.

From my experimenting, if it isn't called, `get_syntax_info()` will return info about the last line/col that was passed to `syn_get_id()`.  If I comment out that line, run `:help printf-d` (has a few concealed characters), then press `w`, everything's skipped up to the next blank line.  I assume it's because of the first character on the line is concealed?

Am I missing something?